docker/run.sh

```diff
@@ -0,0 +1,12 @@
+# For systems without journald.
+mkdir -p /var/log/journal
+
+# Copy host libsystemd into image to avoid compatibility issues. This operation
+# is only performed if the host's lib is mounted into the container through a 
+# volume such as -v /usr/lib64/:/host/lib/.
+if [ ! -z "$(ls /host/lib/libsystemd* 2>/dev/null)" ]; then
+  rm /lib/x86_64-linux-gnu/libsystemd*
+  cp -a /host/lib/libsystemd* /lib/x86_64-linux-gnu/
```

Comment:
Would it make sense to use `ln -s` instead of `cp -a`?

Reply:
My only concern here is that if for whatever reason, the user replaced the contents of those files, that it might adversely affect the host / other containers running on the host. It's a nice way to decouple from the host by using cp. WDYT?

Reply:
You mean inside the container or outside of it? If the former, don't we control the container? And if the user layered their own commands on top of it, are we responsible? If the latter, wouldn't that break the rest of the system as well?
Besides, are we sure that `libsystemd` is statically linked? Even if it is, there are still dependencies on threads and the C runtime, both of which may be different in the host and the container, and may be updated in the same fashion as libsystemd itself while the container is running. This makes me anxious...

Reply:
Just to be clear, this comment wasn't in favor of `ln` vs `cp` -- it's a general concern about using the host's libraries in the first place. But I guess we don't really have much choice, and given the dependencies, `ln` is at least faster than `cp`.

Reply:
Using host's libraries seems inevitable if we want fluentd to collect journal logs [1]. And from our (COS) opinion, collecting journal log is an important feature. We have been using this "cp libsystemd*" to collect journal logs in the old fluentd-gcp image for a while. We want to start using stackdriver-logging-agent as part of b/67426002, but will only be able to do so when jounal logs are collected by fluentd.
I think the base line is that mounting host libraries into the container is inevitable. As they are needed by fluentd plugins. It's not scalable to support all versions of systemd libraries in stackdriver-logging-agent container, which will end up forking it (which is what fluentd-docker-image did in the end [1]). That leaves us the only option of mounting host libs.
I think the question is whether/how do we ensure all dependencies [2] of host's libsystemd are properly mounted into he container, and do not conflict with fluentd within the container. This is a more complicated problem, I think it should be solved later. Since mounting host libs works for now (GKE and COS have been using this for a long time), we could proceed with this for now, just to unblock other things. Later we could evaluate this approach on major OS distros, and fix the mounting and dependency problem then.
WDYT? @igorpeshansky @bmoyles0117 

[1] https://github.com/fluent/fluentd-docker-image/issues/49
[2] http://rpmfind.net/linux/RPM/mageia/6/armv5tl/media/core/updates/libsystemd0-230-12.1.mga6.armv5tl.html

Reply:
Yes, this is fine for now. I'm wondering if a comment to that effect would help, but won't hold this PR for that.